pkgs/development/libraries/mesa/default.nix

```diff
@@ -92,6 +92,11 @@
     "overlay"
     "intel-nullhw"
   ]
+
+# This is ignored when freedreno is not being built. (e.g. on x86_64)
+# virtio is needed for a guest to use DRM native context.
+, freedrenoKernelModeDrivers ? ["msm" "virtio"]
```

Comment:
Does this need to be an option?

Reply:
The full list of available options is `'msm', 'kgsl', 'virtio', 'wsl'`, though not sure if `kgsl` or `wsl` are actually implemented or useful. Did not want to make it difficult for someone adventurous to enable those if they want to experiment.

But we could just hardcode `(lib.mesonOption "freedreno-kmds" "msm,virtio")` if you prefer that, I don't have a problem with that.

Reply:
What's the cost to just build all of them?

Reply:
I guess nothing other than a bit of extra code size. Pushed an update to enable them all unconditionally.

Reply:
I guess next question then: will this break with Mesa 24.3?

Reply:
No idea, tried to ask in #freedreno on IRC but did not get a response to this particular question. I personally would be surprised though if they broke this any time soon, it was at least confirmed to me that all 4 options have some use-case (though I was told that WSL requires some extra tooling to work beside just enabling this), so even if they ever remove one of them I expect it to come with an ample deprecation warning.

Reply:
Next time, please just check the build files. https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/meson_options.txt?ref_type=heads#L222